examples/chat/src/routes/users.rs

```diff
@@ -0,0 +1,67 @@
+use cookie::{Cookie, SameSite};
+use diesel::QueryDsl;
+use diesel_async::RunQueryDsl;
+use via::{Payload, Response};
+
+use crate::models::user::*;
+use crate::{Chat, Next, Request, SESSION};
+
+fn set_current_user(response: &mut Response, state: &Chat, user: &User) -> via::Result<()> {
+    let mut jar = response.cookies_mut().private_mut(state.secret());
+    let cookie = Cookie::build((SESSION, serde_json::to_string(&user)?))
+        .http_only(true)
+        .path("/")
+        .same_site(SameSite::Strict)
+        .secure(true);
+
+    jar.add(cookie);
+
+    Ok(())
+}
+
+pub async fn index(_: Request, _: Next) -> via::Result {
+    todo!()
+}
+
+pub async fn create(request: Request, _: Next) -> via::Result {
+    let (head, future) = request.into_future();
+    let params = future.await?.serde_json::<NewUser>()?;
+
+    let mut connection = head.state().pool().get().await?;
+    let user = User::create(&mut connection, params).await?;
+
+    let mut response = Response::build().status(201).json(&user)?;
+    set_current_user(&mut response, head.state(), &user)?;
+
+    Ok(response)
+}
+
+pub async fn login(request: Request, _: Next) -> via::Result {
+    let (head, future) = request.into_future();
+    let params = future.await?.serde_json::<LoginParams>()?;
+    let user = User::query()
+        .filter(by_username(&params.username))
+        .first(&mut head.state().pool().get().await?)
+        .await?;
```

Comment:
Leaving the door open for now for a password free option that works out of the box. Worst case scenario, I'll save passwords in the database and put comments above every line of code that references a password reminding that an alternative should be preferred in production.